drivers/HwWatchdog.h

```diff
@@ -0,0 +1,94 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_WATCHDOG_H
+#define MBED_WATCHDOG_H
+
+#ifdef DEVICE_WATCHDOG
+
+#include "watchdog_api.h"
+#include "mbed_error.h"
+#include "mbed.h"
+#include "platform/mbed_critical.h"
+#include "LowPowerTicker.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** \addtogroup drivers */
+/** A system timer that will reset the system in the case of system failures or
+ *  malfunctions.
+ *
+ * Example:
+ * @code
+ *
+ * start();
+ *
+ * while (true) {
+ *    wait(0.3);
+ *
+ * }
+ * @endcode
+ * @ingroup drivers
+ */
+
+/** Start an independent watchdog timer
+ *
+ *
+ *  @return status MBED_SUCCESS if the watchdog timer was started
+ *                 successfully. MBED_ERROR_INVALID_ARGUMENT if one of the input
+ *                 parameters is out of range for the current platform.
+ *                 MBED_ERROR_INITIALIZATION_FAILED if one of the enabled input
+ *                 parameters is not supported by the current platform.
+ */
+mbed_error_status_t start();
```

Comment:
Since these functions are in the global namespace you'll want to prefix them so they don't conflict with user code which may be using common names such as `start` and `stop`. If you rename this file as mentioned above you could name this something like `mbed_wdog_management_start()`.